sympy/concrete/zeilberger.py

```diff
@@ -192,13 +191,12 @@ def zb_sum(F, k_a_b, J = 1):
 
     F = F.rewrite(gamma)
 
-    if len(b.free_symbols) != 1:
-        return None
-
-    n = b.free_symbols.pop()
-
-    if not (n.is_positive and n.is_integer) or (
-        hypersimp(F, k) is None or hypersimp(F, n) is None):
+    for w in F.free_symbols.intersection(b.free_symbols.union(a.free_symbols)):
+        n = w
+        if  (n.is_positive and n.is_integer) and not (
```

Comment:
It's generally an antipattern to require assumptions to be set. Instead just check that they aren't False and allow None.